### PR TITLE
ci: add downgrade script for algoliasearch v4 and steps

### PR DESCRIPTION
**Pull request by aymeric-giraudet**
**Summary**

#### [FX-2929](https://algolia.atlassian.net/browse/FX-2929)

**Result**

We run the same tests and type checking as for v5 as they were written for v4 anyway, it does work

[FX-2929]: https://algolia.atlassian.net/browse/FX-2929?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/legacy/downgrade-algoliasearch-v4.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+shell.sed(
+  '-i',
+  /"algoliasearch": ".*"(,)?/,
+  '"algoliasearch": "4.23.2"$1',
+  ...shell.ls('examples/*/*/package.json')
+);
+
+shell.exec('yarn install');
+
+shell.exec('cp -r node_modules/algoliasearch-v4 node_modules/algoliasearch');
```
Comment (aymeric-giraudet) [inline scripts/legacy/downgrade-algoliasearch-v4.js:67]: Struggled with yarn which still keeps `4.23.2` with the `algoliasearch-v4` name, even if we remove the alias.

It does output some warnings like "warning Pattern is trying to unpack in the same destination" which may be related.

Copying it manually works though so idk if we should bother

Comment (Haroenv) [reply scripts/legacy/downgrade-algoliasearch-v4.js:67]: I wonder if this is because of Create InstantSearch App

File: .circleci/config.yml [hunk 2]
```diff
@@ -221,6 +225,25 @@ jobs:
...
+      - *install_yarn_version
+      - *restore_yarn_cache
+      - *run_yarn_install
+      - run:
+          name: Update dependencies
+          command: |
+            ./scripts/legacy/downgrade-algoliasearch-v4.js
+      - run:
+          name: Unit & Integration tests
+          command: yarn run test
```
Comment (Haroenv) [inline .circleci/config.yml:242]: we need to store the test results like in unit tests

**Review by Haroenv — COMMENTED** [review]
overall this looks good, let's just get CI to pass once then lgtm!

File: .circleci/config.yml (hunk 2 above)
Comment (aymeric-giraudet) [reply .circleci/config.yml]: Done and everything passes ! Thanks !

**Review by Haroenv — COMMENTED** [review]
approve :)

**Pull request merged**